docs/odh/oci-model-storage.md

```diff
@@ -8,15 +8,228 @@ which also explains how to deploy models from OCI images.
+  available in Hugging Face. This is a generative AI model.
+* The second example uses the [MobileNet v2-7 model](https://github.com/onnx/models/tree/main/validated/vision/classification/mobilenet)
+  is used. This is a predictive AI model in ONNX format.
 
-The ODH projects provides configurations for the OpenVINO model server, which
+## Creating and deploying an OCI image of IBM's Granite-3.0-2B-Instruct model
+
+IBM's Granite-3.0-2B-Instruct model is [available at Hugging Face](https://huggingface.co/ibm-granite/granite-3.0-2b-instruct).
+To create an OCI container image, the model needs to be downloaded and copied into
+the container. Once the OCI image is built and published in a registry, it can be
+deployed on the cluster.
+
+The ODH project provides configurations for the vLLM model server, which
+supports running the Granite model. Thus, this guide will use this model server
+to demonstrate how to deploy the Granite model stored in an OCI image.
+
+### Storing the Granite model in an OCI image
+
+To download the Granite model, the [`huggingface-cli download` command](https://huggingface.co/docs/huggingface_hub/guides/cli#huggingface-cli-download) can be used.
+The process is as follows:
+* Install the huggingface CLI
+* Use huggingface CLI to download the model
+* Create the final OCI using the downloaded model
+
+This process is implemented in the following multi-stage container build. Create a file named
+`Containerfile` with the following contents:
+```Dockerfile
+##### Stage 1: Download the model
+FROM registry.access.redhat.com/ubi9/python-312:latest as downloader
+
+# Install huggingface-cli
+RUN pip install "huggingface_hub[cli]"
+
+# Download the model
+ARG repo_id
+RUN mkdir models && huggingface-cli download --quiet --max-workers 2 --local-dir ./models $repo_id
+
+##### Stage 2: Build the final OCI model container
+FROM registry.access.redhat.com/ubi9/ubi-micro:latest as model
+
+# Copy from the download stage
+COPY --from=downloader --chown=1001:0 /opt/app-root/src/models /models
+
+# Set proper privileges for KServe
+RUN chmod -R a=rX /models
+
+# Use non-root user as default
+USER 1001
+```
+
+> [!TIP]
+> This Containerfile should be generic enough to download and containerize any
+> model from Hugging Face. However, it has only been tested with the Granite model.
+> Feel free to try it with any other model that can work with the vLLM server.
+
+Notice that model files are copied into `/models` inside the final container. KServe
+expects this path to exist in the OCI image and also expects the model files to
+be inside it.
+
+Also, notice that `ubi9-micro` is used as a base container of the final image.
+Empty images, like `scratch` cannot be used, because KServe needs to configure the model image
+with a command to keep it alive and ensure the model files remain available in
+the pod. Thus, it is required to use a base image that provides a shell.
```

Comment:
thanks @danielezonca for the loop! yes, absolutely because indeed using `FROM scratch` is doable as this proved: https://github.com/tarilabs/demo20241108-base?tab=readme-ov-file#a-minimal-base-image-for-kserve-modelcarsidecar-puposes-that-does-nothing
but likely will need a bit more refinement. 

When this PR is merged, I will provide a new PR on top of this

🚀

Reply:
I think for what regards KServe, it should keep be agnostic of Busybox Vs Ubi-Micro Vs _this example_ etc.

I concur is possibly better however to have _this base image example_ in ODH, before adding it to the Guide

Reply:
✍️  https://github.com/tarilabs/demo20241108-base/issues/11